roles/openshift_master/tasks/systemd_units.yml

```diff
@@ -38,6 +38,11 @@
   when: create_ha_unit_files | changed
 # end workaround for missing systemd unit files
 
+- name: Preserve Master API Proxy Config options
+  command: grep PROXY /etc/sysconfig/{{ openshift.common.service_type }}-master-api
+  register: master_api_proxy
+  when: openshift.master.ha is defined and openshift.master.ha | bool and openshift_master_cluster_method == "native"
```

Comment:
Well, isn't that fun...  In short, no, yes and yes. I would like to see us start getting away from persisting values in openshift_facts where it makes sense...  maybe it doesn't make sense in this case, which would mean that parsing and re-adding it each time is the only solution, but it still seems very error prone that way.